cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -588,6 +589,27 @@ func (ht *hcsTask) DeleteExec(ctx context.Context, eid string) (int, uint32, tim
 	case shimExecStateRunning:
 		return 0, 0, time.Time{}, newExecInvalidStateError(ht.id, eid, state, "delete")
 	}
+
+	if eid == "" {
```

Comment:
If we are cleaning up resources here when deleting the init task (since this triggers `ht.close()`), should we also delete everything in `ht.execs` here?

Reply:
Good question. Let me dig a bit deeper and see if the entire ```hcsTask``` object is not reaped somewhere after init dies and the init task gets deleted.

Reply:
I updated the PR to remove the task object after successfully deleting the init exec.